### PR TITLE
Fix/901/excluding last building error b

### LINKED ISSUE #901: Excluding the last building on the map results in an error, reincluding the buildings makes them all the same size
# Bug
Console Error after last building is excluded, Error in Building Size and Color afterwards if buildings are included again
## Expected Behavior
When the user excludes the last building the map is empty, reincluded buildings look the same as before exclusion.
## Actual Behavior
When the user excludes the last building on the map a console error is thrown, reincluding buidlings afterwards results in all of them being color, due to the metric switching to unary
## Steps to Reproduce the Problem
  1. Exclude all buildings
  2.  The Web Application throws a TypeError: "reduce of empty array with no initial value"
  3. Reincluding buildings makes them all red, metric switches to unary

## Specifications
  - OS: Windows
  - Browser: Firefox, Chrome

### PR DESCRIPTION
**Pull request by Jacob-Kirimi**
excluding last building error

Excluding the last building on the map results in an error, reincluding the buildings makes them all the same size.Console Error after last building is excluded, Difference in Building Size and Color afterwards if buildings are included again due to change  of the original metrics.

closes #901

Expected Behavior
When the user excludes the last building the map is empty, reincluded buildings look the same as before exclusion.

Actual Behavior
When the user excludes the last building on the map a console error is thrown, reincluding buidlings afterwards results in all of them being the same color, due to the metrics switching to unary

### CODE REVIEW

**Comment by Jacob-Kirimi** [comment]
@BridgeAR  remaining is a small clarification on the behaviour once all is excluded. to disable search input or eliminate a condition that is not fullfilled if someone tries to search at this point or alternatively do both.

File: visualization/app/codeCharta/ui/searchBar/searchBar.po.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		await page.waitForFunction(
+			(text: string) => (document.querySelector("#searchInput") as HTMLInputElement).value.includes(text),
+			{},
+			search
+		)
+
+		await clickButtonOnPageElement("#blacklistMenu")
+
+		/* TODO remove timeout */
+		await page.waitForTimeout(500)
```
Comment (NearW) [inline visualization/app/codeCharta/ui/searchBar/searchBar.po.ts:18]:
```suggestion
		await page.waitForSelector("#blacklistMenu.md-clickable")
```

File: visualization/app/codeCharta/ui/dialog/dialog.globalSettings.po.ts [hunk 2]
```diff
@@ -18,6 +18,13 @@ export class DialogGlobalSettingsPageObject {
 		await page.click("div.md-dialog-content layout-selection-component div md-input-container md-select")
 		await page.waitForSelector(".md-select-menu-container.md-active", { visible: true })
 		await page.click('md-select-menu md-content [value="TreeMapStreet"]')
+		await page.waitForSelector("#maxTreeMapFiles-slider > div", { visible: true })
+	}
+
+	async changeLayoutToSquarifiedTreeMap() {
+		await page.click("div.md-dialog-content layout-selection-component div md-input-container md-select")
+		await page.waitForSelector(".md-select-menu-container.md-active", { visible: true })
```
Comment (NearW) [inline visualization/app/codeCharta/ui/dialog/dialog.globalSettings.po.ts:26]:
```suggestion
		await page.waitForSelector(".md-select-menu-container.md-clickable")
```

clickable is better than md-active.

Comment (Jacob-Kirimi) [reply visualization/app/codeCharta/ui/dialog/dialog.globalSettings.po.ts]: done

**Comment by BridgeAR** [comment]
Supersedes #1152

File: visualization/app/codeCharta/util/dataMocks.ts [hunk 3]
```diff
@@ -79,6 +79,47 @@ export const VALID_NODE: CodeMapNode = {
...
+	isExcluded: true,
+	isFlattened: false,
+	children: [
+		{
+			name: "big leaf",
+			type: NodeType.FILE,
+			attributes: { rloc: 100, functions: 10, mcc: 1 },
+			isExcluded: false,
+			isFlattened: false,
+			link: "http://www.google.de"
```
Comment (Med-H) [inline visualization/app/codeCharta/util/dataMocks.ts:95]:
```suggestion
			link: "https://www.google.de"
```

**Review by Med-H — CHANGES_REQUESTED** [review]

**Comment by BridgeAR** [comment]
Due to the many conflicts it's easier to close this and to open a new PR to address this issue. Thanks for the great work though!

**Pull request closed without merging**